src/framework/components/element/text-element.js

```diff
@@ -1160,6 +1185,169 @@ Object.assign(pc, function () {
         }
     });
 
+    Object.defineProperty(TextElement.prototype, "outlineColor", {
+        get: function () {
+            return this._outlineColor;
+        },
+
+        set: function (value) {
+            var r = value.r;
+            var g = value.g;
+            var b = value.b;
+
+            // #ifdef DEBUG
+            if (this._outlineColor === value) {
+                console.warn("Setting element.outlineColor to itself will have no effect");
+            }
+            // #endif
+
+            if (this._outlineColor.r === r && this._outlineColor.g === g && this._outlineColor.b === b) {
+                return;
+            }
+
+            this._outlineColor.r = r;
+            this._outlineColor.g = g;
+            this._outlineColor.b = b;
+
+            if (this._model) {
+                this._outlineColorUniform[0] = this._outlineColor.r;
+                this._outlineColorUniform[1] = this._outlineColor.g;
+                this._outlineColorUniform[2] = this._outlineColor.b;
+
+                for (var i = 0, len = this._model.meshInstances.length; i < len; i++) {
+                    var mi = this._model.meshInstances[i];
+                    mi.setParameter("outline_color", this._outlineColorUniform);
+                }
+            }
+        }
+    });
+
+    Object.defineProperty(TextElement.prototype, "outlineOpacity", {
+        get: function () {
+            return this._outlineColor.a;
+        },
+
+        set: function (value) {
+            if (this._outlineColor.a === value) return;
+
+            this._outlineColor.a = value;
+
+            if (this._model) {
+                this._outlineColorUniform[3] = this._outlineColor.a;
+                for (var i = 0, len = this._model.meshInstances.length; i < len; i++) {
+                    var mi = this._model.meshInstances[i];
+                    mi.setParameter('outline_color', this._outlineColorUniform);
+                }
+            }
+        }
+    });
+
+    Object.defineProperty(TextElement.prototype, "outlineDistance", {
```

Comment:
Should this be called outlineWidth or outlineThickness instead? Distance seems a little less obvious.

Reply:
Also, what are the units on this? Are they the same as for shadowOffset?

Reply:
More generally, API ref docs for the new properties need to be added. 😄